### PR TITLE
system: prevent mypy from checking windows-related method

### PR DESCRIPTION
**Pull request by pared**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

mypy fails on non-windows machines because the `sys` module does not have `getwindowsversion`.

### CODE REVIEW

**Comment by efiop** [comment]
@karajan1001 has also ran into this https://discord.com/channels/485586884165107732/565699007037571084/798846655935152158

File: dvc/system.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 if (
     platform.system() == "Windows"
```
Comment (skshetry) [inline dvc/system.py:13]: @pared, what happens if we change this to following?
```suggestion
    sys.platform == "win32"
```

Does that work?

Comment (skshetry) [reply dvc/system.py:13]: But I am fine with this change as well. 🙂

Comment (pared) [reply dvc/system.py:13]: it does, will change the PR

**Review by skshetry — APPROVED** [review]

**Pull request merged**